internal/btf/core_test.go

```diff
@@ -0,0 +1,134 @@
+package btf
+
+import "testing"
+
+func TestCoreAreTypesCompatible(t *testing.T) {
+	tests := []struct {
+		a, b       Type
+		compatible bool
+	}{
+		{&Void{}, &Void{}, true},
+		{&Struct{Name: "a"}, &Struct{Name: "b"}, true},
```

Comment:
Perhaps add a couple more negative tests? For example, mismatched types (Struct vs Union).

Reply:
Struct and union is already there (is that what you were looking for?), but I'll add some negative tests for the qualifiers.

I also realised that anonymous fields aren't covered at the moment.